### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by leonardoeng13**
- [x] Made sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- [x] Followed SOLID principles.
- [x] Paid special attention to the Single Responsiblity Principle.
  - [x] Make sure that the role of main.rb is only to start the code and ask the user for the initial choice of options (or to quit).
  - [x] Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: library/people/main.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      puts "Person created successfully\n"
+    when '2'
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Specialization: '
+      specialization = gets.chomp
+
+      teacher = Teacher.new(age, name, specialization)
```
Comment (mikethreels) [inline library/people/main.rb]: - [x] Great job on refactoring your code 👏 however in this method try to follow best practices thought in [this solid lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid), pay extra attention to the single responsibility rule. this method is printing, gathering information from the user and creating a new person based on choices made. try to cut this down into smaller steps

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: library/people/main.rb (hunk 1 above)
Comment (leonardoeng13) [reply library/people/main.rb]: Sorry, but I don't understand your request. I'm in pain, I got surgery this last weekend and I can't be sitting here all day. enlight me, please.

**Review by mke2111 — APPROVED** [review]
### Hi @leonardoeng13, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
**Congratulations! 🎉** 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**